### ISSUE TITLE
Lint/RedundantSplatExpansion altering code in unsafe manner

### ISSUE THREAD

**Issue by iHiD** [rubocop-hq/rubocop#9507]
Hello. Thanks for your work on Rubocop. 

I have an issue where `Lint/RedundantSplatExpansion` is incorrectly changing code.

```ruby
# Example 1
# This is (correctly) left unchanged:
method = :some_method
args = Array.new(args, 'a')
res = send(method, *args)

# Example 2
# This is the same code with the variable inlined
method = :some_method
res = send(method, *Array.new(args, 'a'))

# Rubocop incorrectly changes Example 2 to:
method = :some_method
res = Array.new(args, 'a')
```

I can reproduce by pasting that code block into a file and seeing it change as it's linted.

## RuboCop version

```
➜ bundle exec rubocop -V
1.7.0 (using Parser 2.7.2.0, rubocop-ast 1.4.0, running on ruby 2.6.6 x86_64-darwin19)
  - rubocop-minitest 0.10.2
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
```

**Comment by iHiD** [comment]
Thanks.
